project/Http4sPlugin.scala

```diff
@@ -227,6 +227,73 @@ object Http4sPlugin extends AutoPlugin {
     )
   }
 
+  def sbtghactionsSettings: Seq[Setting[_]] = {
+    import sbtghactions._
+    import sbtghactions.GenerativeKeys._
+
+    val setupHugoStep = WorkflowStep.Run(List("""
+      |echo "$HOME/bin" > $GITHUB_PATH
+      |HUGO_VERSION=0.26 scripts/install-hugo
+    """.stripMargin), name = Some("Setup Hugo"))
+
+    def siteBuildStep(subproject: String) = WorkflowStep.Sbt(
+      List(s"$subproject/makeSite"),
+      name = Some(s"Build $subproject"),
+      cond = Some("startsWith(matrix.scala, '2.12.')")
+    )
+
+    def sitePublishStep(subproject: String) = WorkflowStep.Run(List(s"""
+      |eval "$$(ssh-agent -s)"
+      |echo "$$SSH_PRIVATE_KEY" | ssh-add -
+      |git config --global user.name "GitHub Actions CI"
+      |git config --global user.email "ghactions@invalid"
+      |sbt ++$scala_212 $subproject/makeSite $subproject/ghpagesPushSite
+      |
+      """.stripMargin),
+      name = Some("Pubish site: $subproject"),
+      env = Map("SSH_PRIVATE_KEY" -> "${{ secrets.SSH_PRIVATE_KEY }}")
+    )
+
+    inThisBuild(Http4sOrgPlugin.githubActionsSettings ++ Seq(
+      githubWorkflowBuild := Seq(
+        WorkflowStep
+          .Sbt(List("scalafmtCheckAll"), name = Some("Check formatting")),
+        WorkflowStep.Sbt(List("headerCheck", "test:headerCheck"), name = Some("Check headers")),
+        WorkflowStep.Sbt(List("test:compile"), name = Some("Compile")),
+        // TODO reenable
+        // WorkflowStep.Sbt(List("mimaReportBinaryIssues"), name = Some("Check binary compatibility")),
+        WorkflowStep.Sbt(
+          List("unusedCompileDependenciesTest"),
```

Comment:
I'd still like to turn on the other test eventually, but that's a separate ticket.